### PR TITLE
Support for both Brightway 2 and 2.5

### PR DESCRIPTION
**Pull request by cmutel**
🔨 Work in progress 🔨

Factors out Brightway functionality, and imports the correct function depending on which version is installed.

* To install the 2.5 version via pip: `pip install premise[bw25]`.
* On `conda` we might be able to do the same variant, or just create two different packages.

To support this, I moved from `setup.py` to `pyproject.toml`. This also gives us:

* The new standard packaging pipeline 
* Automatic building of wheels (run `python -m build`)
* One place to define the version: `__init__.py`
* One place to define the requirements: `requirements.txt`
* Inclusion of pytest and other config data (`pytest.ini` is removed)

Note that 2.5 support requires `bw2data 4.0.DEV32`, which isn't yet released (will be in ~1 hour from this comment).

We need some testing before this should be released. I have tested the 2.5 functionality, and it builds a matrix and gives a sensible result. However, it would be nice to have a known good result, and test both BW2 and BW2.5 scores. @romainsacchi Could you give a known good result please?

# TODO
- [x] Add notification for `bw2data` version imported, and bw2/2.5
- [x] Test export to normal DB on BW2
- [x] Test export to normal DB on BW2.5
- [ ] Check packaging still works on CI

### CODE REVIEW

**Comment by cmutel** [comment]
There also might be some packaging weirdness, it seems to work for me now but `pyproject.toml` is a harsh mistress...

**Comment by cmutel** [comment]
Also, I don't know anything about how `write_superstructure_db_to_brightway` works, this might be broken? Not sure why it would be but ¯\\\_(ツ)\_/¯

**Comment by tngTUDOR** [comment]
> 🔨 Work in progress 🔨
> 
> Factors out Brightway functionality, and imports the correct function depending on which version is installed.
> 
>     * To install the 2.5 version via pip: `pip install premise[bw25]`.
> 
>     * On `conda` we might be able to do the same variant, or just create two different packages.
> 
By experience, it's "easier" to produce a different conda package (see [the ui package building](https://github.com/brightway-lca/brightway2-ui/blob/main/.github/workflows/conda-package-deploy.yaml) )

File: premise/ecoinvent_modification.py [hunk 1]
```diff
@@ -49,9 +49,16 @@
...
     print_version,
     warning_about_biogenic_co2,
-    write_brightway2_database,
 )
 
+try:
+    import bw_processing
+
+    from .brightway25 import write_brightway_database
+except ImportError:
```
Comment (tngTUDOR) [inline premise/ecoinvent_modification.py:63]: suggestion (non-blocking): It would be useful to warn the user if the import goes OK or (bw2) or wrong.

**Comment by tngTUDOR** [comment]
> There also might be some packaging weirdness, it seems to work for me now but `pyproject.toml` is a harsh mistress...

main branch has some fixes for the github actions to work fine now for packaging pypi pkgs and upload them to the registry.

File: premise/ecoinvent_modification.py (hunk 1 above)
Comment (cmutel) [reply premise/ecoinvent_modification.py:63]: Added to `todo`

**Comment by romainsacchi** [comment]
@cmutel the folowing

`````
from premise import *
from datapackage import Package
bw.projects.set_current("ei39")

scenarios = [
    {"model": "image", "pathway":"SSP2-Base", "year": 2050},
    {"model": "remind", "pathway":"SSP2-PkBudg500", "year": 2050},
]

ndb = NewDatabase(
        scenarios = scenarios,        
        source_db="ecoinvent 3.9.1 cutoff",
        source_version="3.9",
        key='tUePmX_S5B8ieZkkM7WUU2CnO8SmShwmAeWK9x2rTFo=',
)
ndb.update_all()
ndb.write_db_to_brightway()

FU = [
    {act: 1} for db in [
        "ecoinvent 3.9.1 cutoff",
        "ecoinvent_cutoff_3.9_image_SSP2-Base_2050",
        "ecoinvent_cutoff_3.9_remind_SSP2-PkBudg500_2050"
    ] for act in bw2data.Database(db) 
    if act["name"]=="transport, passenger car, electric"
    and act["location"] == "GLO"
]

method = [('IPCC 2021', 'climate change: fossil', 'global warming potential (GWP100)')]

bw2calc.multi_lca.calculation_setups["BEVs"] = {
    "inv": FU,
    "ia": method
}

lca = bw2calc.MultiLCA("BEVs")

print(lca.results)
`````

should return:

`````
[[0.22982712]
 [0.20379977]
 [0.03638609]]
`````

**Comment by romainsacchi** [comment]
> Also, I don't know anything about how `write_superstructure_db_to_brightway` works, this might be broken? Not sure why it would be but ¯\_(ツ)_/¯

`.write_superstructure_db_to_brightway` eventually writes a regular database. So if `.write_db_to_brightway` works, that should work too.

**Comment by cmutel** [comment]
2.5 Scores look good:

```
0.22973739152327347 0.22982712
0.20376602910863864 0.20379977
0.0363075968642278 0.03638609
```

**Comment by cmutel** [comment]
Brightway 2 results are also good!

```
0.2298271180989315 0.22982712 
0.2038174193431219 0.20379977 
0.03630759719576193 0.03638609
```

**Comment by cmutel** [comment]
@romainsacchi I think this is ready to merge. We can push a release like `1.8.0.dev1` and see if it build correctly.

**Pull request merged**